eng/scripts/Language-Settings.ps1

```diff
@@ -172,58 +172,110 @@ function Get-javascript-GithubIoDocIndex()
   GenerateDocfxTocContent -tocContent $tocContent -lang "JavaScript"
 }
 
-# Updates a js CI configuration json.
-# For "latest", we simply set a target package name
-# For "preview", we add @next to the target package name
-function Update-javascript-CIConfig($pkgs, $ciRepo, $locationInDocRepo, $monikerId = $null)
-{
-  $pkgJsonLoc = (Join-Path -Path $ciRepo -ChildPath $locationInDocRepo)
-
-  if (-not (Test-Path $pkgJsonLoc))
-  {
-    Write-Error "Unable to locate package json at location $pkgJsonLoc, exiting."
-    exit(1)
+# "@azure/package-name@1.2.3" -> "@azure/package-name"
+function Get-PackageNameFromDocsMsConfig($DocsConfigName) { 
+  if ($DocsConfigName -match '^(?<pkgName>.+?)(?<pkgVersion>@.+)?$') { 
+    return $Matches['pkgName']
   }
 
-  $allJson = Get-Content $pkgJsonLoc | ConvertFrom-Json
+  throw "Cannot find package name in supplied parameter ($DocsConfigName)"
+}
 
-  $visibleInCI = @{}
+# Given the name of a package (possibly of the form "@azure/package-name@1.2.3")
+# return a package name with the version specified in $packageVersion
+# "@azure/package-name@1.2.3" "1.3.0" -> "@azure/package-name@1.3.0"
+function Get-DocsMsPackageName($packageName, $packageVersion) { 
+  return "$(Get-PackageNameFromDocsMsConfig $packageName)@$packageVersion"
+}
 
-  for ($i = 0; $i -lt $allJson.npm_package_sources.Length; $i++)
-  {
-    $pkgDef = $allJson.npm_package_sources[$i]
-    $accessor = ($pkgDef.name).Replace("`@next", "")
-    $visibleInCI[$accessor] = $i
-  }
+function Update-javascript-DocsMsPackages($DocsRepoLocation) {
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'ci-configs/packages-preview.json') `
+    'preview' 
 
-  foreach ($releasingPkg in $pkgs)
-  {
-    $name = $releasingPkg.PackageId
+  UpdateDocsMsPackages `
+    (Join-Path $DocsRepoLocation 'ci-configs/packages-latest.json') `
+    'latest'
+}
 
-    if ($releasingPkg.IsPrerelease)
-    {
-      $name += "`@next"
+function UpdateDocsMsPackages($DocConfigFile, $Mode) {
+  $publishedPackages = (Get-CSVMetadata).Where({ $_.New -eq 'true' -and $_.Hide -ne 'true' })
```

Comment:
Minor but you might want to share this package list and pass it into the function. I think that will be more important once you start updating these based on the daily build numbers.

Reply:
Given this logic should be common enough I can see it going in at the level of `eng/common/scripts/Update-DocsMsPackages.ps1` ... I think our eventual work in grouped ToC would require domain-specific implementations again but that won't land for a while.

Reply:
"domain-specific implementation" meaning that we'll be rolling out grouped ToC one repo at a time. 

Metadata could take on a different form given that we're not targeting a single file (e.g. packages-latest.json) but could instead have multiple files. However, the details of this implementation are all up in the air at this time. If we have to get domain-specific we'll do that later. 

I can move this up into the level of eng/common and pass the data down to the appropriate places.